### ISSUE TITLE
ember build is not working on ember -cli v2.11.1

### ISSUE THREAD

**Issue by swidjaja** [ember-cli/ember-cli#6783]
Thank you for taking the time to open an issue!

For bug reports please include the following section in your issue details.
If you include instructions on how to reproduce the bug or a failing test case
it will make it easier for us to track down the issue you're having.

---

Output from `ember version --verbose && npm --version`:
```
npm --version
2.15.11

ember version --verbose
ember-cli: 2.11.1
http_parser: 2.7.0
node: 4.6.2
v8: 4.5.103.42
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 56.1
modules: 46
openssl: 1.0.2j
os: darwin x64
```

**Steps to reproduce**
1. Run ember build
2. Ember build will fail with the following stack trace
```

cleaning up...
Build failed.
The Broccoli Plugin: [SimpleConcatConcat: Concat: Vendor Styles/assets/vendor.css] failed with:
Error: Concat: Result is empty and allowNone is falsy.
    at Concat._doPatchBasedBuild (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/concat.js:143:11)
    at Concat.build (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/concat.js:104:17)
    at /Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/node_modules/broccoli-plugin/read_compat.js:93:34
    at tryCatch (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:553:13)
    at publish (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:521:7)
    at flush (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2373:5)
    at nextTickCallbackWith0Args (node.js:436:9)
    at process._tickCallback (node.js:365:13)

The broccoli plugin was instantiated at:
    at Concat.Plugin (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/node_modules/broccoli-plugin/index.js:7:31)
    at new Concat (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/concat.js:38:10)
    at module.exports (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/broccoli-concat/index.js:26:10)
    at EmberAddon.EmberApp.concatFiles (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/broccoli/ember-app.js:369:10)
    at EmberAddon.EmberApp._concatFiles (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/broccoli/ember-app.js:382:15)
    at EmberAddon.EmberApp.styles (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/broccoli/ember-app.js:1334:28)
    at EmberAddon.EmberApp.toArray (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/broccoli/ember-app.js:1647:10)
    at EmberAddon.EmberApp.toTree (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/broccoli/ember-app.js:1668:30)
    at module.exports (/Users/XXXXXXXX/dev/XXXXXXXX/ember-cli-build.js:17:14)
    at CoreObject.module.exports.Task.extend.setupBroccoliBuilder (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/models/builder.js:72:19)
    at CoreObject.module.exports.Task.extend.init (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/models/builder.js:52:10)
    at CoreObject.superWrapper [as init] (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/core-object/lib/assign-properties.js:32:18)
    at CoreObject.Class (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/core-object/core-object.js:32:33)
    at CoreObject.module.exports.Task.extend.run (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at CoreObject.<anonymous> (/Users/XXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/lib/commands/test.js:164:27)
    at tryCatch (/Users/XXXXXXXX/dev/XXXXXXXX/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:538:12)

```

**Here is how the ember-cli-build.js looks like**
```
/*jshint node:true*/
/* global require, module */
var EmberAddon = require('ember-cli/lib/broccoli/ember-addon');

module.exports = function(defaults) {
  var app = new EmberAddon(defaults, {
    // Add options here
  });

  /*
    This build file specifies the options for the dummy test app of this
    addon, located in `/tests/dummy`
    This build file does *not* influence how the addon or the app using it
    behave. You most likely want to be modifying `./index.js` or app's build file
  */

  return app.toTree();
};
```

**package.json**

```
{
  "name": "XXXXXXXXX",
  "version": "0.0.1",
  "description": "XXXXXXXXX XXXXXXXX",
  "keywords": [
    "ember-addon"
  ],
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "repository": "",
  "engineStrict": true,
  "engines": {
    "node": ">= 4.6.2"
  },
  "scripts": {
    "build": "ember build",
    "start": "ember server",
    "test": "ember try:each"
  },
  "dependencies": {
    "ember-cli-babel": "^5.1.7",
    "ember-cli-htmlbars": "^1.1.0",
    "ember-cli-sass": "5.6.0"
  },
  "devDependencies": {
    "babel-preset-es2015": "^6.16.0",
    "babel-register": "^6.18.0",
    "broccoli-asset-rev": "^2.4.5",
    "camelcase": "^4.0.0",
    "chai": "^4.0.0-canary.1",
    "ember-ajax": "^2.4.1",
    "ember-cli": "2.11.1",
    "ember-cli-app-version": "^2.0.0",
    "ember-cli-chai": "0.3.2",
    "ember-cli-code-coverage": "0.3.8",
    "ember-cli-dependency-checker": "^1.3.0",
    "ember-cli-eslint": "3.0.0",
    "ember-cli-htmlbars-inline-precompile": "^0.3.3",
    "ember-cli-inject-live-reload": "^1.4.1",
    "ember-cli-mirage": "0.2.4",
    "ember-cli-mocha": "^0.13.0",
    "ember-cli-release": "^0.2.9",
    "ember-cli-sri": "^2.1.0",
    "ember-cli-test-loader": "^1.1.0",
    "ember-cli-uglify": "^1.2.0",
    "ember-disable-prototype-extensions": "^1.1.0",
    "ember-engines": "0.5.0-beta.3",
    "ember-export-application-global": "^1.0.5",
    "ember-load-initializers": "^0.5.1",
    "ember-resolver": "^2.0.3",
    "ember-sinon": "0.5.1",
    "eslint": "^3.11.1",
    "eslint-config-airbnb-base": "^10.0.1",
    "eslint-plugin-import": "^2.2.0",
    "extend": "^3.0.0",
    "fs-extra": "^2.0.0",
    "grunt-keepalive": "^1.0.0",
    "keepalive": "^3.0.0",
    "loader.js": "^4.0.10",
    "mocha": "^3.2.0",
    "request": "^2.79.0",
    "selenium-webdriver": "^2.53.3",
    "webdriverio": "4.6.1",
    "yargs": "^6.6.0"
  },
  "ember-addon": {
    "configPath": "tests/dummy/config"
  }
}
```

**Comment by rwjblue** [comment]
This is a bug in recently released broccoli-concat (see https://github.com/broccolijs/broccoli-concat/issues/106).

Keeping this open until the fix is out for visibility though...

**Comment by rwjblue** [comment]
I released the code from broccoli-concat@3.1.1 as broccoli-concat@3.2.1 to clear any issues for folks while we get it fixed properly upstream. 

Sorry for the troubles!
